### PR TITLE
Remove bluebird.using

### PR DESCRIPTION
**Pull request by maximelkin**

### CODE REVIEW

**Comment by maximelkin** [comment]
I dont understand acquireConnection implementation:
1) in base it declared with arguments (client, config, txid)
2) also it called in same file with same arguments
3) this method also overriden in mssql, oracle, oracledb, but they have only one argument - config
4) also where are calls to Client.prototype.acquireConnection, but this not so interesting

**Comment by kibertoad** [comment]
@maximelkin Probably it works almost by accident.
See in mssql, for an example:

```
const configConnection = config && config.connection; // this will fail to resolve because client is passed instead of a config
<...>
      try {
        resolve(
          (t.outerTx ? t.outerTx.conn : null) ||
            configConnection ||
            t.client.acquireConnection()
        );
      }
// we will resolve this with a client mistakenly marked as configConnection 
<...>
conn.tx_ = conn.transaction(); // luckily for us, `client.js` has `transaction()` method defined, so probably that works and the whole monster works
```

Frankly, I think this is wrong and should be refactored ruthlessly.

File: lib/runner.js [hunk 1]
```diff
@@ -79,9 +79,7 @@ Object.assign(Runner.prototype, {
     const stream = new PassThrough({ objectMode: true });
```
Comment (kibertoad) [inline lib/runner.js]: One `using` was missed at line 57.

Comment (kibertoad) [reply lib/runner.js]: Oh, wait, sorry, you only changed `runner.js`, `transaction.js` remains as-is.

**Comment by kibertoad** [comment]
@maximelkin Opened https://github.com/knex/knex/pull/3554 to make sure that code is at least not lying - it could actually be that `client.connection` is what was intended to be consumed in the first place, so I didn't touch actual implementation yet, since it seems to be working. If CI passes, I'll merge it so that it's less confusing.

**Comment by kibertoad** [comment]
@maximelkin #3554 was merged, you can rebase to de-confuse code :D

File: lib/transaction.js [hunk 2]
```diff
@@ -208,32 +205,32 @@ class Transaction extends EventEmitter {
   // Acquire a connection and create a disposer - either using the one passed
   // via config or getting one off the client. The disposer will be called once
   // the original promise is marked completed.
-  acquireConnection(client, config, txid) {
+  acquireConnection(config, cb) {
```
Comment (kibertoad) [inline lib/transaction.js]: why get rid of `txid`? it was always passed from this.txid anyway?

Comment (maximelkin) [reply lib/transaction.js]: Yes

Comment (maximelkin) [reply lib/transaction.js]: https://github.com/knex/knex/pull/3552/files#diff-e652b5ccb2708f628a9de325cf105fb7R19

File: lib/runner.js [hunk 3]
```diff
@@ -22,7 +22,7 @@ Object.assign(Runner.prototype, {
   run() {
     const runner = this;
     return (
-      Bluebird.using(this.ensureConnection(), function(connection) {
+      this.ensureConnection(function(connection) {
```
Comment (maximelkin) [inline lib/runner.js]: return native promise, conversion to bluebird promise done in lib/interface.js

File: lib/dialects/mssql/transaction.js [hunk 4]
```diff
@@ -60,43 +60,47 @@ module.exports = class Transaction_MSSQL extends Transaction {
...
         );
       } catch (e) {
         reject(e);
       }
     })
-      .then(function(conn) {
-        conn.__knexTxId = t.txid;
-        if (!t.outerTx) {
-          t.conn = conn;
+      .then((conn) => {
```
Comment (kibertoad) [inline lib/dialects/mssql/transaction.js]: would be awesome to convert those `then` chains into async/await, but separate PR would be better for that.

**Comment by kibertoad** [comment]
Looks fantastic, thank you!

**Pull request merged**